payload/game/ui/Section.cc

```diff
@@ -496,6 +496,57 @@ void Section::logDebuggingInfo(bool verbose) {
             m_pages[i]->logControlsDebug();
         }
     }
+    for (int i = static_cast<int>(PageId::EXT_MININCLUSIVE__);
+            i < static_cast<int>(PageId::EXT_MAXEXCLUSIVE__); ++i) {
+        logPageInfo(m_pages[i]);
+        if (m_pages[i]) {
+            m_pages[i]->logControlsDebug();
+        }
+    }
+}
+
+extern "C" void Page_unfocusMaybe(Page *);
+
+void Section::deinit() {
+    // Since we zero out the arrays, we can trust REPLACED(deinit) does not double-free.
+    for (u32 i = m_activePageCount; i > 0; --i) {
+        auto &page = m_activePages[i - 1];
+        Page_unfocusMaybe(page);
+        page->deactivate();
+        page = nullptr;
+    }
+    m_activePageCount = 0;
+    for (auto &page : m_pages) {
+        if (page == nullptr) {
```

Comment:
```suggestion
    for (auto *&page : m_pages) {
        if (!page) {
```
Along with similar changes in this function.

I think an implicit nullptr check is fine in this context as there is no confusion - the page either exists or it doesn't.

Reply:
I think I prefer `nullptr`

Reply:
Should we at least have `auto *&` for transparency about the type?

Reply:
Yes